controllers/grafanadashboard/dashboard_pipeline.go

```diff
@@ -119,10 +120,13 @@ func (r *DashboardPipelineImpl) validateJson() error {
 }
 
 func (r *DashboardPipelineImpl) shouldUseContentCache() bool {
-	cacheDuration := r.Dashboard.Spec.ContentCacheDuration.Duration
-	contentTimeStamp := r.Dashboard.Status.ContentTimestamp
+	if r.Dashboard.Status.ContentGzip != nil && r.Dashboard.Spec.ContentCacheDuration != nil && r.Dashboard.Status.ContentTimestamp != nil {
+		cacheDuration := r.Dashboard.Spec.ContentCacheDuration.Duration
+		contentTimeStamp := r.Dashboard.Status.ContentTimestamp
 
-	return cacheDuration > 0 && contentTimeStamp.Add(cacheDuration).After(time.Now())
+		return cacheDuration <= 0 || contentTimeStamp.Add(cacheDuration).After(time.Now())
+	}
+	return false
 }
 
 // Try to get the dashboard json definition either from a provided URL or from the
```

Comment:
As you mention in the docs i grafanaURL is defined json will be ignored. I guess we should update this comment to make it clear.

Reply:
Added some more clarifying comments here